### PR TITLE
[Adreno] Fix mem_scope annotations for prim funcs having several heads

### PR DESCRIPTION
**Pull request by elvin-n**
1) decision if function can work with textures. To remove dependency on the order of traversing, we decide to markup function support textures if any input op support textures [src/relay/transforms/annotate_texture_storage.cc](https://github.com/apache/tvm/compare/main...Deelvin:tvm:amalyshe/adreno_injective?expand=1#diff-409f906bd8fdca700e76726e204f722ae4b8a8ab39376ffb9bb46434d063678c)
2) fix of injective schedules for non blocking case for performance improvements ([python/tvm/topi/adreno/utils.py](https://github.com/apache/tvm/compare/main...Deelvin:tvm:amalyshe/adreno_injective?expand=1#diff-1d21d85eaf5c83c690c54c8041b665759580004bd1d07204c416e5bee6d6329e)

@echuraev

### CODE REVIEW

File: python/tvm/topi/adreno/utils.py [hunk 1]
```diff
@@ -525,28 +525,26 @@ def bind_data_copy(stage, axis_to_vectorize=None):
...
-            stage.bind(fused, te.thread_axis("blockIdx.x"))
-            stage.bind(by, te.thread_axis("blockIdx.y"))
-            stage.bind(ty, te.thread_axis("threadIdx.y"))
+            stage.vectorize(axes[-1])
         else:
-            stage.bind(fused, te.thread_axis("blockIdx.x"))
-            stage.bind(*axes[-1:], te.thread_axis("threadIdx.x"))
+            ftc = numpy.prod(shape)
+            vthread = get_div(ftc, 8)
+            fused = stage.fuse(*[stage.op.axis[i] for i in range(len(stage.op.axis))])
```
[the remark this answers is not in the document]
Comment (elvin-n) [reply python/tvm/topi/adreno/utils.py]: changed

File: python/tvm/topi/adreno/utils.py [hunk 2]
```diff
@@ -525,28 +525,26 @@ def bind_data_copy(stage, axis_to_vectorize=None):
...
+            ftc = ftc / vthread
+            num_thread = get_div(ftc, 1024 // vthread)
```
[the remark this answers is not in the document]
Comment (elvin-n) [reply python/tvm/topi/adreno/utils.py:543]: added comment

File: python/tvm/topi/adreno/utils.py [hunk 3]
```diff
@@ -525,28 +525,27 @@ def bind_data_copy(stage, axis_to_vectorize=None):
...
-            stage.bind(ty, te.thread_axis("threadIdx.y"))
+            stage.vectorize(axes[-1])
         else:
-            stage.bind(fused, te.thread_axis("blockIdx.x"))
-            stage.bind(*axes[-1:], te.thread_axis("threadIdx.x"))
+            ftc = numpy.prod(shape)
+            vthread = get_div(ftc, 8)
+            fused = stage.fuse(*stage.op.axis)
+            ftc = ftc / vthread
+            # Max work group size on the most Adreno GPU
```
Comment (echuraev) [inline python/tvm/topi/adreno/utils.py:542]: I'm not sure, that it is clear that you speak about `1024`. Suggest to write such this:
```suggestion
            # 1024 is max work group size on the most Adreno GPU
```

**Review by echuraev — APPROVED** [review]
LGTM. Thanks!

**Comment by elvin-n** [comment]
@masahi could you please take a look to it?

**Review by masahi — APPROVED** [review]

**Comment by elvin-n** [comment]
@tvm-bot rerun

**Pull request merged**